javascript/ql/lib/semmle/javascript/frameworks/NoSQL.qll

```diff
@@ -623,6 +640,10 @@ private module MarsDB {
 
     string getMethod() { result = method }
 
+    override DataFlow::Node getAResult() {
+      PromiseFlow::loadStep(this.(DataFlow::SourceNode).getALocalUse(), result, Promises::valueProp())
```

Comment:
If you add `DataFlow::CallNode` to the `extends` clause, then you don't need an inline cast here.